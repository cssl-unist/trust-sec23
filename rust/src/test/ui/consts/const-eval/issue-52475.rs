fn main() {
    let _ = [(); {
        let mut x = &0;
        let mut n = 0;
        while n < 5 {
            n = (n + 1) % 5; //~ ERROR evaluation of constant value failed
            x = &0; // Materialize a new AllocId
        }
        0
    }];
}
