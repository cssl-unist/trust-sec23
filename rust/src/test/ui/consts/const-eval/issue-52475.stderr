error[E0080]: evaluation of constant value failed
  --> $DIR/issue-52475.rs:6:17
   |
LL |             n = (n + 1) % 5;
   |                 ^^^^^^^^^^^ exceeded interpreter step limit (see `#[const_eval_limit]`)

error: aborting due to previous error

For more information about this error, try `rustc --explain E0080`.
